doc/api_reference.md

```diff
@@ -465,6 +505,55 @@ Otherwise returns `blocking_kind::maybe`.
 Senders can customise this algorithm by providing an overload of
 `tag_invoke(tag_t<blocking>, const your_sender_type&)`.
 
+## Many Sender Algorithms
+
+### `bulk_transform(ManySender sender, Func func, FuncPolicy policy) -> ManySender`
+
+For each `set_next(values...)` result produced by `sender`, invokes
+`func(values...)` and produces the result of that call as its `set_next()`
+result.
+
+The `policy` argument is optional and if absent, defaults to `get_execution_policy(func)`.
+
+The resulting execution policy incorporates the union of the constraints
+placed on the execution of the function and the execution of the
+downstream receiver's `set_next()` method.
+
+i.e. both the down-stream ManyReceiver's execution policy and the function's
+execution policy must allow parallel execution for the bulk_transform
+operation to permit parallel execution. Same for unsequenced execution.
+
+This algorithm is transparent to `set_value()`, `set_error()` and `set_done()`
+completion signals.
+
+### `bulk_join(ManySender source) -> Sender`
+
+Joins a bulk operation on a ManySender and turns it into a SingleSender
+operation that completes once all of the `set_next()` calls have completed.
+
+The input `source` sender must be a ManySender of `void` (ie. no values passed
+to `set_next()`).
+
+The returned single-sender is transparent to the `set_value()`, `set_error()`
+and `set_done()` signals.
+
+### `bulk_schedule(Scheduler sched, Count n) -> ManySender`
+
+Returns a ManySender of type `Count` that sends the values `0 .. n-1`
+to the receiver's `set_next()` channel.
+
+The default implementation of this algorithm schedules a single
+task onto the specified scheduler using `schedule()` and then calls
+`set_next()` in a loop.
+
+Scheduler types are permitted to customise the `bulk_schedule()` operation
+to allow more efficient implementations. e.g. a thread-pool may choose to
+split the work up into M pieces to execute across M different threads.
+
+Note that customisations must still adhere to the constraints placed on
+valid executions of `set_next()` according to the execution policy returned
+from `get_execution_policy()`.
```

Comment:
I mostly like where you're going here (it's basically the direction I went with this, way back before Cologne), but I think that this sort of shape information is of more general applicability than this, and confining all customizations of scheduling with a given shape to go through this one customization point is probably not sufficient. I think that this information also needs to be a bit stickier—probably via a customization point like `get_shape(ManySender)` that is customized if and only if the shape is known lazily, as is the case with the result of `bulk_schedule`. Does that make sense?